sourcegraph/sourcegraph
# [SG-36513] Accessibility: Code monitoring logs screen reader issues

### Audit type
Screen reader navigation
### User journey audit issue
[#34194](https://github.com/sourcegraph/sourcegraph/issues/34194)
### Problem description
- Buttons to expand should not have aria-label. Instead, the label should be on the expand/collapse icon.
- "Monitor details" and "N new results" links should be outside the button and the icon should read "Open in a new tab"
SUCESS/SKIPPED/ERROR tags should have aria-label so they are not read as uppercase. The label should read ", monitor state: "
- Add hidden "Monitor name:" before every name and "Last run:" before every timestamp on the left column

![image](https://user-images.githubusercontent.com/206864/171750198-7d071c43-00e8-47d1-90b7-75a912f51446.png)

### Expected behavior
- Issues resolved as described above.

### Additional details
[#35151](https://github.com/sourcegraph/sourcegraph/issues/35151) and [#35158](https://github.com/sourcegraph/sourcegraph/issues/35158) are related issues that should probably be fixed first.

### Refs
[SourceGraph Issue](https://github.com/sourcegraph/sourcegraph/issues/36513)
[GitStart Issue](https://app.gitstart.com/clients/sourcegraph/tickets/SG-36513)

### Test Plan
- Navigate to the [Code Monitoring Log](https://client-sourcegraph-frontend.onrender.com/code-monitoring?visible=1) page and ensure you're on the `Logs` tab.
- Inspect the elements using the developer tool to ensure they ascribe to the above `expected behavior`.
- Run the `screen reader` and ensure that the labels are announced/read as required.




## App preview:

- [Web](https://sg-web-contractors-sg-36513.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-zckanzlago.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
